File: awesomestuff/src/main/java/com/awesome/dialog/AwesomeGeneralDialog.java
package com.awesome.dialog;

import android.content.Context;
import android.support.annotation.ColorInt;
import android.support.annotation.ColorRes;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;
import android.widget.Button;


import com.awesome.R;

import static android.view.View.*;

/**
 * Created by yarolegovich on 16.04.2016.
 * If null is passed instead on click listener - dialog will be just closed on click.
 */
@SuppressWarnings("WeakerAccess")
public class AwesomeGeneralDialog extends AwesomeBase<AwesomeGeneralDialog> {

    public static final int POSITIVE_BUTTON = R.id.ld_btn_yes;
    public static final int NEGATIVE_BUTTON = R.id.ld_btn_no;
    public static final int NEUTRAL_BUTTON = R.id.ld_btn_neutral;

    private Button positiveButton;
    private Button negativeButton;
    private Button neutralButton;

    public AwesomeGeneralDialog(Context context) {
        super(context);
    }

    public AwesomeGeneralDialog(Context context, int theme) {
        super(context, theme);
    }

    {
        positiveButton = findView(R.id.ld_btn_yes);
        negativeButton = findView(R.id.ld_btn_no);
        neutralButton = findView(R.id.ld_btn_neutral);
    }

    public AwesomeGeneralDialog setPositiveButton(@StringRes int text, OnClickListener listener) {
        return setPositiveButton(string(text), listener);
    }

    public AwesomeGeneralDialog setPositiveButton(String text, @Nullable OnClickListener listener) {
        positiveButton.setVisibility(VISIBLE);
        positiveButton.setText(text);
        positiveButton.setOnClickListener(new ClickListenerDecorator(listener, true));
        return this;
    }

    public AwesomeGeneralDialog setNegativeButtonText(@StringRes int text) {
        return setNegativeButton(string(text), null);
    }

    public AwesomeGeneralDialog setNegativeButtonText(String text) {
        return setNegativeButton(text, null);
    }

    public AwesomeGeneralDialog setNegativeButton(@StringRes int text, OnClickListener listener) {
        return setNegativeButton(string(text), listener);
    }

    public AwesomeGeneralDialog setNegativeButton(String text, @Nullable OnClickListener listener) {
        negativeButton.setVisibility(VISIBLE);
        negativeButton.setText(text);
        negativeButton.setOnClickListener(new ClickListenerDecorator(listener, true));
        return this;
    }

    public AwesomeGeneralDialog setNeutralButtonText(@StringRes int text) {
        return setNeutralButton(string(text), null);
    }

    public AwesomeGeneralDialog setNeutralButtonText(String text) {
        return setNeutralButton(text, null);
    }

    public AwesomeGeneralDialog setNeutralButton(@StringRes int text, @Nullable OnClickListener listener) {
        return setNeutralButton(string(text), listener);
    }

    public AwesomeGeneralDialog setNeutralButton(String text, @Nullable OnClickListener listener) {
        neutralButton.setVisibility(VISIBLE);
        neutralButton.setText(text);
        neutralButton.setOnClickListener(new ClickListenerDecorator(listener, true));
        return this;
    }

    public AwesomeGeneralDialog setButtonsColor(@ColorInt int color) {
        positiveButton.setTextColor(color);
        negativeButton.setTextColor(color);
        neutralButton.setTextColor(color);
        return this;
    }

    public AwesomeGeneralDialog setButtonsColorRes(@ColorRes int colorRes) {
        return setButtonsColor(color(colorRes));
    }

    public AwesomeGeneralDialog setOnButtonClickListener(OnClickListener listener) {
        return setOnButtonClickListener(true, listener);
    }

    public AwesomeGeneralDialog setOnButtonClickListener(boolean closeOnClick, OnClickListener listener) {
        OnClickListener clickHandler = new ClickListenerDecorator(listener, closeOnClick);
        positiveButton.setOnClickListener(clickHandler);
        neutralButton.setOnClickListener(clickHandler);
        negativeButton.setOnClickListener(clickHandler);
        return this;
    }

    public AwesomeGeneralDialog setPositiveButtonText(@StringRes int text) {
        return setPositiveButton(string(text), null);
    }

    public AwesomeGeneralDialog setPositiveButtonText(String text) {
        return setPositiveButton(text, null);
    }

    public AwesomeGeneralDialog setPositiveButtonColor(@ColorInt int color) {
        positiveButton.setTextColor(color);
        return this;
    }

    public AwesomeGeneralDialog setNegativeButtonColor(@ColorInt int color) {
        negativeButton.setTextColor(color);
        return this;
    }

    public AwesomeGeneralDialog setNeutralButtonColor(@ColorInt int color) {
        neutralButton.setTextColor(color);
        return this;
    }

    public AwesomeGeneralDialog setPositiveButtonColorRes(@ColorRes int colorRes) {
        return setPositiveButtonColor(color(colorRes));
    }

    public AwesomeGeneralDialog setNegativeButtonColorRes(@ColorRes int colorRes) {
        return setNegativeButtonColor(color(colorRes));
    }

    public AwesomeGeneralDialog setNeutralButtonColorRes(@ColorRes int colorRes) {
        return setNeutralButtonColor(color(colorRes));
    }

    @Override
    protected int getLayout() {
        return R.layout.dialog_standard;
    }
}

File: fusion/src/main/java/com/awesome/dialog/AwesomeGeneralDialog.kt
package com.awesome.dialog

import android.content.Context
import android.support.annotation.ColorInt
import android.support.annotation.ColorRes
import android.support.annotation.StringRes
import android.widget.Button


import com.awesome.R

import android.view.View.*

/**
 * Created by Sandip Savaliya on 16.04.2016.
 * If null is passed instead on click listener - dialog will be just closed on click.
 */
class AwesomeGeneralDialog : AwesomeBase<AwesomeGeneralDialog> {

    private var positiveButton: Button? = null
    private var negativeButton: Button? = null
    private var neutralButton: Button? = null

    override val layout: Int
        get() = R.layout.dialog_standard

    constructor(context: Context) : super(context)

    constructor(context: Context, theme: Int) : super(context, theme)

    init {
        positiveButton = findView(R.id.ld_btn_yes)
        negativeButton = findView(R.id.ld_btn_no)
        neutralButton = findView(R.id.ld_btn_neutral)
    }

    fun setPositiveButton(@StringRes text: Int, listener: OnClickListener): AwesomeGeneralDialog {
        return setPositiveButton(string(text), listener)
    }

    fun setPositiveButton(text: String, listener: OnClickListener?): AwesomeGeneralDialog {
        positiveButton!!.visibility = VISIBLE
        positiveButton!!.text = text
        positiveButton!!.setOnClickListener(AwesomeBase.ClickListenerDecorator(listener, true, dialog!!))
        return this
    }

    fun setNegativeButtonText(@StringRes text: Int): AwesomeGeneralDialog {
        return setNegativeButton(string(text), null)
    }

    fun setNegativeButtonText(text: String): AwesomeGeneralDialog {
        return setNegativeButton(text, null)
    }

    fun setNegativeButton(@StringRes text: Int, listener: OnClickListener): AwesomeGeneralDialog {
        return setNegativeButton(string(text), listener)
    }

    fun setNegativeButton(text: String, listener: OnClickListener?): AwesomeGeneralDialog {
        negativeButton!!.visibility = VISIBLE
        negativeButton!!.text = text
        negativeButton!!.setOnClickListener(AwesomeBase.ClickListenerDecorator(listener, true, dialog!!))
        return this
    }

    fun setNeutralButtonText(@StringRes text: Int): AwesomeGeneralDialog {
        return setNeutralButton(string(text), null)
    }

    fun setNeutralButtonText(text: String): AwesomeGeneralDialog {
        return setNeutralButton(text, null)
    }

    fun setNeutralButton(@StringRes text: Int, listener: OnClickListener?): AwesomeGeneralDialog {
        return setNeutralButton(string(text), listener)
    }

    fun setNeutralButton(text: String, listener: OnClickListener?): AwesomeGeneralDialog {
        neutralButton!!.visibility = VISIBLE
        neutralButton!!.text = text
        neutralButton!!.setOnClickListener(AwesomeBase.ClickListenerDecorator(listener, true, dialog!!))
        return this
    }

    fun setButtonsColor(@ColorInt color: Int): AwesomeGeneralDialog {
        positiveButton!!.setTextColor(color)
        negativeButton!!.setTextColor(color)
        neutralButton!!.setTextColor(color)
        return this
    }

    fun setButtonsColorRes(@ColorRes colorRes: Int): AwesomeGeneralDialog {
        return setButtonsColor(color(colorRes))
    }

    fun setOnButtonClickListener(listener: OnClickListener): AwesomeGeneralDialog {
        return setOnButtonClickListener(true, listener)
    }

    fun setOnButtonClickListener(closeOnClick: Boolean, listener: OnClickListener): AwesomeGeneralDialog {
        val clickHandler = AwesomeBase.ClickListenerDecorator(listener, closeOnClick, dialog!!)
        positiveButton!!.setOnClickListener(clickHandler)
        neutralButton!!.setOnClickListener(clickHandler)
        negativeButton!!.setOnClickListener(clickHandler)
        return this
    }

    fun setPositiveButtonText(@StringRes text: Int): AwesomeGeneralDialog {
        return setPositiveButton(string(text), null)
    }

    fun setPositiveButtonText(text: String): AwesomeGeneralDialog {
        return setPositiveButton(text, null)
    }

    fun setPositiveButtonColor(@ColorInt color: Int): AwesomeGeneralDialog {
        positiveButton!!.setTextColor(color)
        return this
    }

    fun setNegativeButtonColor(@ColorInt color: Int): AwesomeGeneralDialog {
        negativeButton!!.setTextColor(color)
        return this
    }

    fun setNeutralButtonColor(@ColorInt color: Int): AwesomeGeneralDialog {
        neutralButton!!.setTextColor(color)
        return this
    }

    fun setPositiveButtonColorRes(@ColorRes colorRes: Int): AwesomeGeneralDialog {
        return setPositiveButtonColor(color(colorRes))
    }

    fun setNegativeButtonColorRes(@ColorRes colorRes: Int): AwesomeGeneralDialog {
        return setNegativeButtonColor(color(colorRes))
    }

    fun setNeutralButtonColorRes(@ColorRes colorRes: Int): AwesomeGeneralDialog {
        return setNeutralButtonColor(color(colorRes))
    }

    companion object {

        val POSITIVE_BUTTON = R.id.ld_btn_yes
        val NEGATIVE_BUTTON = R.id.ld_btn_no
        val NEUTRAL_BUTTON = R.id.ld_btn_neutral
    }
}
